src/main/src/main/java/org/geoserver/security/SecuredResourceNameChangeListener.java

```diff
@@ -91,61 +89,75 @@ public void handleModifyEvent(CatalogModifyEvent event) throws CatalogException
 
     @Override
     public void handlePostModifyEvent(CatalogPostModifyEvent event) throws CatalogException {
-        // no names where change
-        if (event.getPropertyNames().indexOf("name") == -1) return;
-        String oldName =
-                String.valueOf(event.getOldValues().get(event.getPropertyNames().indexOf("name")));
-        String newName =
-                String.valueOf(event.getNewValues().get(event.getPropertyNames().indexOf("name")));
-        // dont go anything if name has not changed
-        if (oldName.equalsIgnoreCase(newName)) return;
-        // will be used if name of
-        String resourceWorkSpaceName = null;
-        CatalogInfo resourceInfo = null;
-        // if workspace has been renamed
-        if (event.getSource() instanceof WorkspaceInfo) {
-            // if a security rule exists for the workspaec whose name has been changed
-            if (workspaceHasSecurityRule(oldName)) {
-                LOGGER.info(
-                        String.format(
-                                "Updating Security Rules for Renamed Workspace: %s", oldName));
+        final String oldName;
+        final String newName;
+        {
+            final int indexOfName = event.getPropertyNames().indexOf("name");
+            // no names where changed
+            if (indexOfName == -1) return;
+            oldName = String.valueOf(event.getOldValues().get(indexOfName));
+            newName = String.valueOf(event.getNewValues().get(indexOfName));
+            // dont go anything if name has not changed
+            if (oldName.equalsIgnoreCase(newName)) return;
+        }
 
-                // get rules in which workspace name should be updated
-                List<DataAccessRule> rulesToModifyList =
-                        getDataAccessRule(event.getSource(), oldName);
-                for (DataAccessRule rule : rulesToModifyList) rule.setRoot(newName);
-            }
-        } else if (event.getSource() instanceof ResourceInfo) {
+        final CatalogInfo eventSource = event.getSource();
+        final Predicate<DataAccessRule> filter;
+        final Consumer<DataAccessRule> updater;
+        if (eventSource instanceof WorkspaceInfo) {
+            // if workspace has been renamed
+            filter = filter(oldName::equalsIgnoreCase, l -> true);
```

Comment:
Wouldn't this catch also global layer groups? A workspace and a global layer group might share the same name.

Reply:
That is a good catch, thanks for looking into this. In fact, they shouldn't collide, since workspace rules never have `null` layer, but an actual name or the `*` glob. e.g., if you have both a ws and a global group named `osm`, rules on them would be like this:

![image](https://user-images.githubusercontent.com/207423/151821536-d3b4930b-dcaf-4d22-bef2-1b7af3132868.png)

I'm adding a fix and a test case.

Reply:
@aaime in writing the test case, I found that when a workspace is removed, its data rules aren't removed.
do you know if that's intentional or should they also be removed?
cause I'm just at the point where it'd be a couple lines of code to fix it

Reply:
Does not seem intentional to me, more like another issue.